src/storage/copier/column_chunk.cpp

```diff
@@ -158,6 +178,24 @@ void ColumnChunk::resize(uint64_t numValues) {
     }
 }
 
+void ColumnChunk::populateWithDefaultVal(common::ValueVector* defaultValueVector) {
+    defaultValueVector->setState(std::make_shared<common::DataChunkState>());
+    auto valPos = defaultValueVector->state->selVector->selectedPositions[0];
+    defaultValueVector->state->selVector->resetSelectorToValuePosBufferWithSize(
+        common::DEFAULT_VECTOR_CAPACITY);
+    for (auto i = 0u; i < defaultValueVector->state->selVector->selectedSize; i++) {
+        defaultValueVector->state->selVector->selectedPositions[i] = valPos;
+    }
+    auto numValuesAppended = 0u;
+    while (numValuesAppended < common::StorageConstants::NODE_GROUP_SIZE) {
+        auto numValuesToAppend = std::min(common::DEFAULT_VECTOR_CAPACITY,
+            common::StorageConstants::NODE_GROUP_SIZE - numValuesAppended);
+        defaultValueVector->state->selVector->selectedSize = numValuesToAppend;
```

Comment:
If you want to go through `isUnfiltered` branch in `copyVectorToBuffer` and `append` functions, check if we can set the vector to unfiltered here. As it should be unfiltered and optimized without going through selected positions.